tests/Builders/ComponentParameterBuilderTests.cs

```diff
@@ -0,0 +1,326 @@
+﻿using System;
+using System.Threading.Tasks;
+using Bunit.SampleComponents;
+using Microsoft.AspNetCore.Components;
+using Shouldly;
+using Xunit;
+
+namespace Bunit
+{
+    public class ComponentParameterBuilderTests
+    {
+        private readonly ComponentParameterBuilder<AllTypesOfParams<string>> _sut;
+
+        public ComponentParameterBuilderTests()
+        {
+            _sut = new ComponentParameterBuilder<AllTypesOfParams<string>>();
+        }
+
+        [Fact(DisplayName = "Add CascadingParameter (nullable integer) and Build")]
+        public void Test001()
+        {
+            // Arrange
+            const int value = 42;
+
+            // Arrange
+            _sut.Add(c => c.NamedCascadingValue, value);
+            var result = _sut.Build();
+
+            // Assert
+            result.Count.ShouldBe(1);
+
+            var parameter = result[0];
+            parameter.IsCascadingValue.ShouldBeTrue();
+            parameter.Name.ShouldBe(nameof(AllTypesOfParams<string>.NamedCascadingValue));
+            parameter.Value.ShouldBe(value);
+        }

```

Comment:
```suggestion
        public void Test001()
        {
            // Arrange
            var sut = CreateSut();
            const int value = 42;

            // Arrange
            sut.Add(c => c.NamedCascadingValue, value);
            var result = sut.Build();

            // Assert
            result.Count.ShouldBe(1);

            var parameter = result[0];
            parameter.IsCascadingValue.ShouldBeTrue();
            parameter.Name.ShouldBe(nameof(AllTypesOfParams<string>.NamedCascadingValue));
            parameter.Value.ShouldBe(value);
        }
```

This small change should be duplicated to each test.

Reply:
```
var sut = CreateSut();
```

I can create this helper method, however my personal preference is just add this `new` code in the constructor from the test-class because the testclass is created each time by xunit. See also https://xunit.net/docs/shared-context.html#constructor

Reply:
Yep, I am aware of how xunit work in this regard, and I have been using the pattern you are using in many other tests ive written previously. But I like to use a factory method instead of the constructor and a class field as it keeps all parts of the test local to the test method, and avoids forcing the user to scroll up/down to see what is going on. The factory method still allows us to refactor the construction of SUT without breaking all tests, so in my mind, it is worth having one extra line in each test for the added clarity.

Reply:
OK. Fixed.